sample/system.py

```diff
@@ -1,60 +1,120 @@
-            '5c0bb1d2-bf1b-497a-be10-63adc7b5c6a2',
-            'fe0e32ec-ca7f-4604-b6e5-5f59a85d1450']
+from elasticsearch import Elasticsearch
 
-link = "https://api.arxivdigest.org/"
+from init_index import init_index
+from index import run_indexing
 
+API_KEY = '4c02e337-c94b-48b6-b30e-0c06839c81e6'
+API_URL = 'http://127.0.0.1:5000/'
+INDEX = 'main_index'
 
-def users(start, api_key):
-    req = request.Request('%susers?from=%d' % (link, start),
+
+def get_user_ids(start, api_key, api_url):
+    """Queries the arXivDigest API for user ids starting from 'start'. """
+    req = request.Request('%susers?from=%d' % (api_url, start),
                           headers={"api_key": api_key})
     resp = request.urlopen(req)
     return json.loads(resp.read())
 
 
-def articles(api_key):
-    req = request.Request('%sarticles' % link, headers={"api_key": api_key})
-    try:
+def get_user_info(user_ids, api_key, api_url, batch_size=100):
+    """Queries the arXivDigest API for userdata for the given 'user_ids'."""
+    user_info = {}
+    for i in range(0, len(user_ids), batch_size):
+        user_id_string = ','.join([str(u) for u in user_ids[i:i + batch_size]])
+
+        req = request.Request('%suserinfo?user_id=%s' % (api_url, user_id_string),
+                              headers={"api_key": api_key})
         resp = request.urlopen(req)
-    except error.HTTPError as e:
-        print(e.read())
-    return json.loads(resp.read())
+        user_info.update(json.loads(resp.read())['userinfo'])
+    return user_info
 
 
-def sendRecommendations(data, api_key):
-    data = json.dumps({'recommendations': data}).encode('utf8')
-    req = request.Request(link+"recommendation", data=data, headers={
-                          'Content-Type': 'application/json', "api_key": api_key})
-    resp = request.urlopen(req)
+def get_articles_by_keywords(keywords, index, window_size=1, size=10000):
+    """Retrieves articles from the Elasticsearch index mentioning 'keywords',
+    the 'window_size' is the number of days back in time articles will be included from."""
+    es = Elasticsearch()
+    query = {
+        'query': {
+            'bool': {
+                'must': {
+                    'match': {
+                        'catch_all': {
+                            'query': keywords,
+                        }
+                    }
+                },
+                'filter': {
+                    'range': {
+                        'date': {
+                            'gte': 'now-{}d'.format(window_size)
+                        }
+                    }
+                }
+            }
+        }
+    }
+    res = es.search(index=index, body=query, params={"size": size})
+    return res
+
+
+def send_recommendations(recommendations, api_key, api_url, batch_size=100):
+    """Sends the recommendations to the arXivDigest API. Recommendations should be a dict of user_id, recommendation pairs."""
```

Comment:
Nit: limit line length to 80 chars (everywhere)

Reply:
We've been using 120 the default pycharm line length guide. We can start using 80 and updating where we work but changing the whole codebase could take some time.

Reply:
120 is fine, don't change then.